cmd/snap-bootstrap/partition/encrypt.go

```diff
@@ -0,0 +1,117 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package partition
+
+import (
+	"bytes"
+	"crypto/rand"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+var (
+	tempFile = ioutil.TempFile
+)
+
+// Our key is 32 bytes long
+const (
+	keySize = 32
+)
+
+type EncryptionKey [keySize]byte
+
+func NewEncryptionKey() (EncryptionKey, error) {
+	var key EncryptionKey
+	_, err := rand.Read(key[:])
```

Comment:
I wonder if we should be paranoid here and do something like:
```
n, err := rand.Read(key[:])
if n != keySize { return fmt.Errorf("cannot get enough data from the random number generator (%v read, %v expected)", n, keySize };
```
just in case the go implementation changes or something unexpected happens.

Reply:
According to the [crypto/rand Read() documentation](https://golang.org/pkg/crypto/rand/#Read) it should be protected against short reads, but being extra cautious won't hurt.

Reply:
Indeed, given that it's even document in the read doc string I guess the extra check is probably overkill.